bitbot/tarot.go

```diff
@@ -0,0 +1,119 @@
+package bitbot
+
+import (
+    "github.com/whyrusleeping/hellabot"
+    "math/rand"
+    "strings"
+    "strconv"
+)
+
+var TarotTrigger = NamedTrigger{
+    ID:   "Tarot",
+    Help: "Request tarot cards, default 1. Usage !tarot [num cards].",
+    Condition: func(irc *hbot.Bot, m *hbot.Message) bool {
+        return m.Command == "PRIVMSG" && strings.HasPrefix(m.Trailing, "!tarot")
+    },
+    Action: func(irc *hbot.Bot, m *hbot.Message) bool {
+        if (len(m.Content) < 7) {
+            resp := tarotCards[rand.Intn(len(magic8responses)-1)]
+            irc.Reply(m, resp)
+        } else {
+            msg := strings.TrimPrefix(m.Content, "!tarot ")
+            if num, err := strconv.Atoi(msg); err == nil {
+                if (num < 1 || num > len(tarotCards)) {
+                    num = 1
+                }
+                deck := rand.Perm(77)
```

Comment:
Normally I'd initialize these closer to the top and put explanatory comments for their usage

```
var (
        deck []int  // random permutation of index values from tarotCards
        card  int  // individual index element pulled from deck
)